jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -448,35 +448,33 @@ private static String parseStringParameter(String paramName, String[] parameters
 
 	private static DiagnosticProperties doJFR(String diagnosticCommand) {
 		DiagnosticProperties result = null;
-
 		// split the command and arguments
 		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
 		IPC.logMessage("doJFR: ", diagnosticCommand);
-
 		// ensure there's at least one part for the command
 		if (parts.length == 0) {
 			return DiagnosticProperties.makeErrorProperties("Error: No JFR command specified");
 		}
-
 		String command = parts[0].trim();
 		String[] parameters = Arrays.copyOfRange(parts, 1, parts.length);
-
 		String fileName = parseStringParameter("filename", parameters, null);
 		IPC.logMessage("doJFR: filename = ", fileName);
-		boolean setFileName = (fileName != null) && !fileName.isEmpty();
-		if (setFileName) {
-			if (!VM.setJFRRecordingFileName(fileName)) {
-				return DiagnosticProperties.makeErrorProperties("setJFRRecordingFileName failed");
-			} else {
-				jfrRecordingFileName = fileName;
-			}
-		}
 
 		try {
 			if (command.equalsIgnoreCase(DIAGNOSTICS_JFR_START)) {
 				if (VM.isJFRRecordingStarted()) {
 					result = DiagnosticProperties.makeErrorProperties("One JFR recording is in progress [" + jfrRecordingFileName + "], only one recording is allowed at a time.");
 				} else {
+					// only JFR.start command is allowed to change the recording filename
+					boolean setFileName = (fileName != null) && !fileName.isEmpty();
+					if (setFileName) {
+						// the recording filename should be set before VM.startJFR() which invokes JFRWriter:openJFRFile()
```

Comment:
@JasonFengJ9 Actually, we should turn the filename string to a path, then convert it to an absolute path. With something like 

```
Path path = Path.of(fileName);
Path parent = path.getParent()
if ( new File(parent.getParent().toString()).exists()) { //output an error message

String filePath = path.toAbsolutePath();
```

Reply:
> if ( new File(parent.getParent().toString()).exists()) { //output an error message

Does this intend to throw an error if there is an existing recording file? Current behaviour is to overwrite it.

Note: this won't fix the [2. When setting a relative file path filename=jfrfile.jfr it should set it in the PWD of the jcmd process, not the pwd of the target process.](https://github.com/eclipse-openj9/openj9/issues/21069) because the code snippet is running within the context of target VM. I am working on a similar code snippet but in `jcmd` context (still in progress).

Reply:
Okay, maybe ignore this suggestion for that PR then. I just wanted to make sure absolute paths work properly.

Reply:
The absolute path works with this PR, turning a relative file path to the `pwd` of the target process needs some work.